lib/matplotlib/widgets.py

```diff
@@ -2967,99 +3033,123 @@ def _onmove(self, event):
         """Motion notify event handler."""
 
         state = self._state | self._default_state
+        rotate = ('rotate' in state and
+                  self._active_handle in self._corner_order)
+        eventpress = self._eventpress
+        # The calculations are done for rotation at zero: we apply inverse
+        # transformation to events except when we rotate and move
+        if not (self._active_handle == 'C' or rotate):
+            inv_tr = self._get_rotation_transform().inverted()
+            event.xdata, event.ydata = inv_tr.transform(
+                [event.xdata, event.ydata])
+            eventpress.xdata, eventpress.ydata = inv_tr.transform(
+                [eventpress.xdata, eventpress.ydata]
+                )
+
+        dx = event.xdata - eventpress.xdata
+        dy = event.ydata - eventpress.ydata
+        refmax = None
+        if 'data_coordinates' in state:
+            refx, refy = dx, dy
+        else:
+            refx = event.xdata / (eventpress.xdata + 1e-6)
```

Comment:
Can you add a comment about the 1e-6 padding here?

Reply:
Are you sure 1e-6 is a universal solution? Can `eventpress.xdata` not become -1e-6?

Maybe better to 

```
xdata_press = eventpress.xdata if eventpress.xdata != 0 else 1-e6
refx = event.xdata / xdata_press
```